### ISSUE TITLE
QR Code to Import Settings Will Not Scan

### ISSUE THREAD

**Issue by crdennett** [thunderbird/thunderbird-android#8464]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App

Thunderbird for Android

### App version

8.0

### Where did you get the app from?

Google Play

### Android version

15

### Device model

Pixel 8 Pro

### Steps to reproduce

1.  On desktop, export to mobile to get QR code screen
2. Open app on phone
3. Click on import settings on splash screen
4. Camera turns on
5. Aim camera at Q code on desktop screen
6. Nothing happens.  

### Expected behavior

At step 6 I would expect the camera to lock onto the QR code and either automatically import the settings or give me a button to do that.

### Actual behavior

Nothing happens when the phone camera is aimed at the QR code.  I tried the built in QR code scanner in the phone and it locks in right away.  It gives me the option to copy the test embedded in the QR code but I can't use that in the app.

**Comment by mnassAN** [comment]
Same behaviour on Andoid 14 - Galaxy A53 - OneUI 6.1 - App version 8

When I use the camera QR Scanner - it says "No app to open the information"

**Comment by kewisch** [comment]
The QR code embeds text only, so a third party app won't do anything special with it. You need to use Thunderbird for Android's account setup camera instead. Was there something about the explanation we could make more clear to users they can't use the a normal QR code scanner app?

Based on the devices you mention it doesn't sound like the camera is lacking, so I'm surprised there is no function. Are you using dark mode or a special color scheme?

**Comment by crdennett** [comment]
I didn't use the QR scanner app at first.   As I noted I clicked the Import  Settings button and it immediately turned on the camera.  I pointed the camera at the QR code but it did not lock on nor was there any button to read the code.  I just tried a QR scanner app to see if the QR code could be read by it.  It was.  I am not using dark mode.  I did see some other bug report about that.  I also tried varying the brightness of my monitor and also switched to a different monitor.  Nothing worked,   So, I simply configured  it manually.  Since I also have the K9 Mail app installed, it was easy to go through the process and get it working.   I figure it's probably something with the phone itself.

**Comment by crdennett** [comment]
Well, I don't thing it's the camera.  Just went to a website where I generated a QR code for my website.  Turned on the camera and pointed it at the code and it locked on and read the code,

**Comment by curzon01** [comment]
I can confirm the issue also on a Motorola g84:
- Thunderbird for Android has permission to access the camera, can display the TB Desktop QR Code picture but cannot scan the TB (also any) QR code
It doesn't matter on which of 3 different monitors I display the TB Desktop QR code, even using different monitor presets or desktop dark/light mode doesn't matter.
- The mobile camera app, on the other hand, does not have any problem with any of the QR Codes displayed on the same different monitors with same different settings (monitor presets, desktop dark or light), scans in and shows me the configuration text correctly.

Extracting the configuration text with the mobile camera app and re-encoding the QR codes (with my desktop app QtQR) with different settings including shortening the content (less complex QR code) does not lead to a different result: TB does not scan any of the codes while the mobile camera app has no problems scanning them.

I suspect the cause is somehow the Android api call to the camera, because
- another Android app (Authenticator) has the same behaviour like TB for Android (displays the QR Code but does not scan any QR code)
- while my bank app does not have the issue.

**Comment by Nionor** [comment]
Same problem here with Pixel 7 on Android 15

**Comment by nchevsky** [comment]
On a Samsung Galaxy S22 Ultra on OneUI 6.1 (Android 14), scanning only works if I export exactly **one account without a password**—the moment I add _either_ (b) a password for the one account _or_ (b) additional accounts (with or without passwords), scanning does nothing. In other words, scanning succeeds only with "small" QR codes and fails with "large" (i.e. higher-resolution) QR codes that contain more data.

**Comment by leubeem** [comment]
> On a Samsung Galaxy S22 Ultra on OneUI 6.1 (Android 14), scanning only works if I export exactly **one account without a password**—the moment I add _either_ (b) a password for that one account _or_ (b) additional accounts (with or without passwords), scanning does nothing.
> 
> In other words, scanning succeeds with "small"/low-res QR codes and fails with "large"/high-res QR codes that contain more data.

Same here with Galaxy S23, OneUI 6.1 (Android 14) and 1.) Thunderbird Mobile Beta v8.0b5 and also 2.) Thunderbird Mobile v8.0

**Comment by jnzd** [comment]
I've had the same issue with Android 15 on a Google Pixel 8 and Thunderbird Desktop set to the default dark theme. I was able to scan all QR codes by changing the theme to light. This seems to be an issue with the QR scanner not working properly with different colors and/or contrast ratios (not that with the light theme it is black on white, but with the dark theme it is white on dark blue or grey).

**Comment by nchevsky** [comment]
> (…) and Thunderbird Desktop set to the default dark theme. I was able to scan all QR codes by changing the theme to light.

Success at last! 🎉 Switching to the light theme alone wasn't enough for me, though; I had to _also_ move my phone considerably farther away from the screen, so that much more of the surroundings were in view. This is counterintuitive at best, as the scanner instructions say to _"align the QR code"_—implying one should get just the code, and nothing else, within the frame.

**Comment by mnassAN** [comment]
Can confirm, scanning far away from the display works for me. Light screen was set all the time for my tries.

**Comment by Nionor** [comment]
> Switching to the light theme alone wasn't enough for me, though; I had to _also_ move my phone considerably farther away from the screen, so that more of the surroundings were in view. 

Worked for me too!

**Comment by curzon01** [comment]
In the meantime, I can confirm the experiences here, but it's very tricky.
Technically it works, but only works for me within a very narrow distance range.
Just wondering why the camera app has no problems at all with dark/light or the distances.

**Comment by SimonZoro02** [comment]
> > Switching to the light theme alone wasn't enough for me, though; I had to _also_ move my phone considerably farther away from the screen, so that more of the surroundings were in view.
> 
> Worked for me too!

Same for me, it's quite bizarre but it worked!! :)

**Comment by batty-h** [comment]
It's not always the camera. Thunderbird for Android seems to have a problem with special Email-account-names. In my case, the QR-code-scan didn't work with an POP3-Gmail-Account I used in the "recent mode" ("recent:_my name_@gmail.com"). After deleting the "recent:" from the account-name and the server-settings on my desktop, the new QR-code became readable (the settings for 7 accounts were imported without any problems)

**Comment by gitrazorhub** [comment]
Same issue for me on Android.

-Desktop TB version 128.4.1esr (64-bit) Ubuntu Mate.
-Android 14
-Pixel 5

After the step 'Select Import settings' the camera opens and a DONE button is shown but pressing it does nothing but move to the next page without any settings transferred.

**Comment by Christopher-Hayes** [comment]
Multiple accounts was the problem for me too. If I only import one account at a time, it works.

**Comment by haubourg** [comment]
Hi, I just did a screenshot and changed contrast settings in a photo editor software. QR code were read immediately. Just using a darker black color instead of this light gray will solve the problem. Is there any reason such a light gray is used when generating the QRcode image ?

**Comment by nielbock** [comment]
I have the same issue with that phone. QR code scanning has never been an issue before. Just this one. I only realizead that the QR Code is unsually complex.

I tried a few fixes reported here. None of them worked. I tried to scan the code simply with my smartphone camera and it was recognized immediately. Just not via the import functionality.

**Comment by nielbock** [comment]
> I had the similar issue and had to follow a similarly step as mentioned above
> 
> > I had to also move my phone considerably farther away from the screen, so that more of the surroundings were in view.
> 
> However, my code was not automatically detected, I just simply pressed `Done` and it grabbed the information. So it seems more of a detection issue than camera reading the QR.

That worked for me, too. Interesting.

**Comment by SKelem** [comment]
The TB app does not read the second QR code (2 of 2).

I was able to have the Android Thunderbird read the first QR code (3 accounts), but it was not able to read the second QR code.

I set my TB theme to Auto/Light, and that allowed me to read the first QR code.

TB app (version 8.1) on Android 14, Samsung Galaxy A53 5G.

**Comment by SzZ-gh** [comment]
The problem I had with 2 accounts was that the Authentication method was not specified under Server Settings. It was empty. Selecting the Password transmitted, insecurely option, QR code regeneration worked. (without this settings, it passed a NULL value in one of the properties, which is why QR reading did not work)

**Comment by manikandancode** [comment]
Yes not working. fix this. thanks

**Comment by osiktech** [comment]
I can confirm, that it worked fine for me, changing the desktop app from dark to light theme. I tested it with Thunderbird 8.2 on my mobile and tablet.

**Comment by soleroks** [comment]
I generated QR for 3 of my accounts and scanned from a bit further, it somehow works this way.

**Comment by indig0F10w** [comment]
**TL;DR FIX - generate codes for each account individually. QR codes containing multiple accounts are not working.**

Ulefone Armor X10 Pro, Android 12, no root, Tunderbird Mobile v8.2 installed via F-Droid. QR generated contains two accounts. Every QR scanner app reads the code immediately, except Thunderbird's Import account via QR function.

Tried switching to light mode, closer/further camera range, portrait/landscape view mode, regenerated QR code, set permissions for the app to allow all... nothing helps. The phone makes a click sound, and nothing happens.

Removed the app from F-Droid and installed the Google Play Store version. Everything is the same.

I tried importing accounts one by one, and this is working flawlessly.

**Comment by kewisch** [comment]
Can you check the text of the generated QR code to see if there is something obviously wrong, like `null` fields?

**Comment by indig0F10w** [comment]
> Can you check the text of the generated QR code to see if there is something obviously wrong, like `null` fields?

It has one null field. I replaced my info.

`[1,[1,1],[0,"imap.gmail.com",993,3,6,"MYACCOUNT1@gmail.com","MYACCOUNT1@gmail.com",""],[[[0,"smtp.gmail.com",465,3,6,"MYACCOUNT1@gmail.com",""],["MYACCOUNT1@gmail.com","MYNAME"]]],[1,"pop.ht.hr",110,2,null,"ACCOUNT_USERNAME","MYEMAIL@ri.ht.hr",""],[[[0,"mail.ht.hr",587,0,1,"ACCOUNT_USERNAME",""],["MYEMAIL@ri.ht.hr","MYNAME"]]]]`

**Comment by kewisch** [comment]
You'll need to update the account settings in Thunderbird Desktop until there is no more null. This is usually the authentication type.

**Comment by indig0F10w** [comment]
> This is usually the authentication type.

The account is configured and if I change it it will not work. Connection security field which is set to "None" in Thunderbird Desktop. This is a Thunderbird setting which is allowed and this error should be handled by Thunderbird mobile by parsing null fields as "none" as it was set on Thunderbird Desktop.

**Comment by kewisch** [comment]
@indig0F10w Can you confirm that the value for "None" is indeed null? To my knowledge this is an integer field so "None" should correspond to 0, and the other values are 1,2,3. The issue that we've seen so far is that due to some unknown reason the value is completely unset on Desktop. On Desktop this is ok likely because it coerces null to 0, but when exporting the data it doesn't adhere to the spec we agreed on. I'm in contact with the desktop folks on this though.

What you can likely do is change each setting that is null to another value and back, then it should work.

**Comment by ErikUden** [comment]
Just wanted to confirm that I had the exact same bug and it was fixed by changing the theme of Thunderbird Desktop to light! After that, it scanned all the codes correctly.

Five other people on Reddit said this is what fixed it for them: https://www.reddit.com/r/Thunderbird/s/Nse6dj2bLG

**Comment by Lars-Schumann** [comment]
> Just wanted to confirm that I had the exact same bug and it was fixed by changing the theme of Thunderbird Desktop to light! After that, it scanned all the codes correctly.
> 
> Five other people on Reddit said this is what fixed it for them: https://www.reddit.com/r/Thunderbird/s/Nse6dj2bLG

This did not solve the problem for me, only sometimes a single account without passwords scans properly, anything more never scans.
Desktop (NixOS 26.05): Mozilla Thunderbird 146.0.1
Mobile (GrapheneOS): version 15.0, installed via PlayStore

**Comment by primis** [comment]
Can confirm, same bug. Multi-account is still broken, Only single account without passwords at a time scans properly for me as well.
I am using a 21" 1080p monitor, so pixel pitch isn't exactly huge, which might have something to do with it.

Desktop (Windows 10 -- Sorry):  Mozilla Firefox 140.7.1esr (64-bit)
Mobile (Android 16 on my Xperia 1 vii): Version 16.1 Mobile installed via F-Droid

**Comment by realrossmanngroup** [comment]
I have the same problem. Using light mode doesn't fix it. Nothing appears to fix it. It sees the QR code, I ca nsee it on screen, I know the android application has permissions, but the thunderbird android app does not recognize a QR code or do anything with it

**Comment by Troublemaker1978** [comment]
Xiaomi 15 Ultra same Problem. Nothing is working, not the single import and not the mass import. Smartphone cam scans without problem

**Comment by nodeblox** [comment]
Xiaomi Redmi Note 15 pro 5g only works without passwords, however for multi-account (without password) too. (light AND dark mode)

**Comment by J1963** [comment]
Same issue here on a pixel 8. Also when just importing one account. Note however that that account has six aliases. On a Pixel 8, a regular QR scanner picks up the code immediately with dark mode on. As that works as expected, the issue must be within Thunderbird.  Note that both on desktop as on phone, the latest versions are installed. Other OS: Windows 11 Pro and Android 16 respectively.

Suggested solution for future development: skip scanning from within the app. Create a QR code, instruct users to use a regular scanning app and develop something that allows for the import of the scanned data. Could also be a separate plugin.

**Comment by eliotlynx** [comment]
I'm experiencing the same bug. My phone (Motorola G31) scans other QR codes with no issues, but it's impossible to scan the Thunderbird code. I made sure Thunderbird has camera permissions on my phone. What I tried:

- using both light and dark mode on my PC,
- increasing and decreasing the size of the QR code on my PC,
- moving closer and away from my PC,
- increasing and decreasing the brightness of my PC screen,
- exporting only one of the email adresses.

Nothing works. I wish there was a way to generate a code instead of the QR code or have the app automatically detect the code instead of needing to press "Done".

**Comment by kdlh6** [comment]
Pixel 8a only works without passwords

**Comment by Kurgan-** [comment]
Had the same issue today on a customer's phone. The QR can be read with a generic QR code reader on the same Phone (Samsung A series) but TB just shows the image and does nothing. It does not read the QR code at all.
